frontend/src/Home.jsx

```diff
@@ -3,42 +3,87 @@ import "./Home.css";
 import SearchBar from "./components/home/SearchBar.jsx";
 import NewBoardModal from "./components/home/NewBoardModal.jsx";
 import KudosList from "./components/home/KudosList.jsx";
+import { getAllBoards } from "./utils/reusedFunctions.js";
 
-const Home = ({ setCurrentBoardID, setCurrentBoardName }) => {
+const ModalOptions = Object.freeze({
+  ALL: "all",
+  RECENT: "recent",
+  CELEBRATION: "celebration",
+  THANK_YOU: "thank-you",
+  INSPIRATION: "inspiration",
+});
+
+const INIT_RELOAD = 0;
+
+const Home = ({
+  setCurrentBoardID,
+  setCurrentBoardName,
+  setToggled,
+  toggled,
+}) => {
   const [isNewBoardModalVisible, setNewBoardModalVisibility] = useState(false);
   const [visibleBoards, setVisibleBoards] = useState([]);
+  const [reload, setReload] = useState(INIT_RELOAD);
 
-  const getBoards = useCallback(async () => {
+  const getBoards = async () => {
     const response = await fetch("http://localhost:3000/boards");
     const data = await response.json();
     setVisibleBoards(data);
-  });
+  };
+
+  async function handleSortChange(event) {
+    event.preventDefault();
+    const response = await fetch(
+      `http://localhost:3000/boards/category/${event.target.value}`
+    );
+    const data = await response.json();
+    setVisibleBoards(data);
+    console.log(data);
+  }
 
   useEffect(() => {
     getBoards();
-  }, [getBoards]);
+  }, [reload]);
+
+  useEffect(() => {
+    document.body.classList.toggle("toggled", toggled); // between light and dark mode
```

Comment:
Why not have a simple click handler and do this within it? `useEffect` is unnecessary.